app/src/main/java/com/github/swent/echo/data/repository/RepositoryImpl.kt

```diff
@@ -45,8 +46,17 @@ class RepositoryImpl(
         }
         val localResult = localDataSource.getAssociation(associationId, ASSOCIATION_CACHE_TTL)
         if (localResult == null) {
-            val remoteResult = remoteDataSource.getAssociation(associationId)
-            if (remoteResult != null) localDataSource.setAssociation(remoteResult)
+            val remoteResult =
+                try {
+                    remoteDataSource.getAssociation(associationId)
+                } catch (e: HttpRequestException) {
+                    return localDataSource.getAssociation(associationId, FETCH_ALL)
```

Comment:
As you are using the try catch block as an expression you can drop the `return` key word (the last statement is automatically returned). You can also do this in all try catch blocks below.

Reply:
Thanks for you feedback. 
Taking a close look on this, I realize that no, this is not the case. The return is intentional here to exit the function.
If you check the line above the try (currently 49), there is a variable assignment done. Based on that variable assignment, further actions are taken. However in case of the exception the result from line 53 needs to be returned directly as otherwise it would be written back to the local database unnecessarily.